### PR TITLE
docs: explain how to find a clang-format patch generated by CI

### LINKED ISSUE #4020: CI: Improve instructions for finding and applying the patch generated by failed Github Actions (Version: 1.8.1)
## Issue Description

The instructions provided on failed jobs ([example](https://github.com/ripple/rippled/runs/4455580047)) don't clearly explain where to find the generated patch file. This can be a problem for anyone who is not familiar with how Github Actions pages are organized.

The general idea is: 
> From the details page, click on "Summary", then scroll down to the "Artifacts" section. There's a clang-format.patch/levelization.patch file there. Download it into your repo then run `git apply [patchfile]`. Commit and push.

### PR DESCRIPTION
**Pull request by intelliot**
Fix #4020

### CODE REVIEW

**Comment by oeggert** [comment]
I'll review it today.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -121,6 +121,16 @@ this:
 You can format individual files in place by running `clang-format -i <file>...`
 from any directory within this project.
 
+There is a Continuous Integration job that runs clang-format on PRs. If the code does not comply, an artifact will be generated which is a patch file that corrects any auto-fixable formatting issues.
```
Comment (oeggert) [inline CONTRIBUTING.md:124]:
```suggestion
There is a Continuous Integration job that runs clang-format on pull requests. If the code doesn't comply, a patch file that corrects auto-fixable formatting issues is generated.
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -121,6 +121,16 @@ this:
...
+
+To download the patch file:
+
+1. Next to `clang-format / check (pull_request) Failing after #s` -> click **Details** to open the details page.
```
Comment (oeggert) [inline CONTRIBUTING.md:128]:
```suggestion
1. Click **Details** next to the `clang-format / check (pull_request) Failing after #s`.
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -121,6 +121,16 @@ this:
...
+2. Left menu -> click **Summary**
```
Comment (oeggert) [inline CONTRIBUTING.md:129]:
```suggestion
2. Click **Summary** from the left menu.
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -121,6 +121,16 @@ this:
...
+3. Scroll down to near the bottom-right under `Artifacts` -> click **clang-format.patch**
```
Comment (oeggert) [inline CONTRIBUTING.md:130]:
```suggestion
3. Click **clang-format.patch** under `Artifacts`.
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -121,6 +121,16 @@ this:
...
+4. Download it to your local git repo and then run `git apply [patch-file-name]`.
```
Comment (oeggert) [inline CONTRIBUTING.md:131]:
```suggestion
4. Download the file to your local git repository and run `git apply [patch-file-name]`.
```

**Review by oeggert — APPROVED** [review]
Approved with suggestions for language.

File: CONTRIBUTING.md (hunk 2 above)
Comment (intelliot) [reply CONTRIBUTING.md:128]: Reading the original text left-to-right, it tells you what to look for - in the order that you need to look for it. In other words, you need to look for `clang-format / check (pull_request) Failing after #s` before you can click **Details**. In the way that you propose, it's as if the steps are reversed.

File: CONTRIBUTING.md (hunk 3 above)
Comment (intelliot) [reply CONTRIBUTING.md:129]: Similarly - you generally need to look for the left menu before you can click **Summary**.

File: CONTRIBUTING.md (hunk 4 above)
Comment (intelliot) [reply CONTRIBUTING.md:130]: Same concern

**Comment by ximinez** [comment]
I think it would also be useful to have these instructions in the "What happened?" step of the clang-format.yml job.

**Review by oeggert — APPROVED** [review]
LGTM

**Comment by intelliot** [comment]
@ximinez is this PR worth merging as-is? If not, let's close it, and you can open a new PR with your preferred changes.

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -119,6 +119,16 @@ this:
...
 from any directory within this project.
 
+There is a Continuous Integration job that runs clang-format on pull requests. If the code doesn't comply, a patch file that corrects auto-fixable formatting issues is generated.
+
+To download the patch file:
+
+1. Next to `clang-format / check (pull_request) Failing after #s` -> click **Details** to open the details page.
+2. Left menu -> click **Summary**
+3. Scroll down to near the bottom-right under `Artifacts` -> click **clang-format.patch**
+4. Download the file to your local git repository and run `git apply [patch-file-name]`.
```
Comment (ximinez) [inline CONTRIBUTING.md:129]: When you click on the artifact, the file that's downloaded is a zip file. So there needs to be another step to unzip the file (or a clause in step 4).

**Review by ximinez — APPROVED** [review]
Other than the one comment, this is fine to merge as-is. We can change the job later in another PR. Contributors will hopefully look here first.

File: CONTRIBUTING.md (hunk 6 above)
Comment (intelliot) [inline CONTRIBUTING.md:129]:
```suggestion
4. Download the zip file and extract it to your local git repository. Run `git apply [patch-file-name]`.
```

**Pull request merged**